### ISSUE TITLE
bip32: Build breakage, possibly due to conflicting generic-array dependencies.

### ISSUE THREAD

**Issue by jnicholls** [iqlusioninc/crates#858]
Rust 1.54

I'm depending upon various crypto projects, which results in an amalgamation of dependencies that conflict. Specifically, generic-array v0.12.4 and v0.14.4 are in the dependency tree, and cargo has successfully been able to work with both of these dependencies and solve applicable conflicts until today. Today, I added the solana-sdk dependency to my project, which caused bip32 to break as seen below.

```
error[E0282]: type annotations needed
  --> /Users/jarrednicholls/.cargo/registry/src/github.com-1ecc6299db9ec823/bip32-0.2.1/src/public_key.rs:71:25
   |
71 |         self.to_bytes().as_ref().try_into().expect("malformed key")
   |         ----------------^^^^^^--
   |         |               |
   |         |               cannot infer type for type parameter `T` declared on the trait `AsRef`
   |         this method call resolves to `&T`
   |
   = note: type must be known at this point

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
error: could not compile `bip32
```

The only explanation I can find after digging through the dependency tree and applicable code, is that cargo is now choosing generic-array 0.12.4 to satisfy the bip32 build instead of generic-array 0.14.4... _or something like that_. i.e. this most recently added dependency was the straw that broke the camel's back, officially confusing cargo or the compiler.

A workaround is to use the `GenericArray::as_slice` method instead of `AsRef<[T]>::as_ref` in that impl of PublicKey, which results in no type confusion and is more clear imho. I'll submit a PR with the suggested change for more discussion.

**Comment by tony-iqlusion** [comment]
Looks like this can be closed now that #859 is merged

**Comment by jnicholls** [comment]
Thanks @tony-iqlusion!
